### DISCUSSION TITLE
Is change to `flake.nix` supposed to rebuild the whole project from scratch (dependencies)?

### DISCUSSION THREAD

**Discussion by dpc** [ipetkov/crane#73] [Q&A] [answered]
I don't know if I'm doing something wrong, but when I work on setting up all sorts of things in `flake.nix`, each time I have to suffer through the complete rebuild. It seems to me that `flake.nix` modifications should not cause that. Am I missing anything?

**Comment by dpc** [comment, marked as answer]
Short answer: if you used `src = ./.;` then yes.

Long answer: to avoid it you need to go back to filtering `src` to include only relevant files. Example: https://github.com/fedimint/fedimint/pull/380/commits/2f6137d75988e82f09e719c32945bd80bad73add

**Reply by ipetkov** [reply]
Glad you figured it out @dpc, that was exactly what I was going to suggest! I've opened #78 to update the FAQ section of the README so hopefully it is clearer to someone else in the future!
